Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm

```diff
@@ -366,6 +365,37 @@ - (void)setImageAnalysis:(CocoaImageAnalysis *)analysis;
 
 #if ENABLE(IMAGE_ANALYSIS)
 
+class ImageAnalysisCallbackAggregator : public ThreadSafeRefCounted<ImageAnalysisCallbackAggregator> {
+public:
+    static Ref<ImageAnalysisCallbackAggregator> create(CompletionHandler<void()>&& completionHandler)
+    {
+        return adoptRef(*new ImageAnalysisCallbackAggregator(WTFMove(completionHandler)));
+    }
+
+    ~ImageAnalysisCallbackAggregator()
+    {
+        if (RunLoop::isMain()) {
```

Comment:
Could have used `ensureOnMainRunLoop` but I recommend I suggestion above ^ instead.

Reply:
(Yep! Going to go with adding the "call on main thread" variant to WTF)